symbolic-debuginfo/src/pdb.rs

```diff
@@ -112,6 +112,117 @@ impl From<pdb_addr2line::Error> for PdbError {
     }
 }
 
+/// Remap function type for VCS-specific path transformations
+type RemapFn = fn(&srcsrv::EvalVarMap) -> Option<String>;
+
+/// Remap function for Perforce SRCSRV entries
+///
+/// Extracts depot path (var3) and changelist (var4), then returns
+/// the path in format: depot/path@changelist
+fn remap_perforce(var_map: &srcsrv::EvalVarMap) -> Option<String> {
+    let depot_path = var_map.get("var3")?;
+    let changelist = var_map.get("var4");
+
+    // Strip leading // from depot path for code mapping compatibility
+    let depot = depot_path.trim_start_matches("//");
+
+    match changelist {
+        Some(cl) if !cl.is_empty() => {
+            // Append changelist as @changelist for Perforce
+            Some(format!("{}@{}", depot, cl))
+        }
+        _ => {
+            // Depot path without changelist
+            Some(depot.to_string())
+        }
+    }
+}
+
+/// Version control systems that we support for SRCSRV path remapping
+/// Maps VCS name (case-insensitive) to its remap function
+fn get_vcs_remap_functions() -> std::collections::HashMap<&'static str, RemapFn> {
+    let mut map = std::collections::HashMap::new();
+    map.insert("perforce", remap_perforce as RemapFn);
+    map
+}
+
+/// Source server mappings extracted from PDB files using the srcsrv crate.
+///
+/// This handles the full SRCSRV format including variable evaluation and
+/// supports extracting changelist information for supported VCS systems.
+#[derive(Clone)]
+struct SourceServerMappings {
```

Comment:
Another option could be to turn this into an enum with a case for each supported VCS, but it's not necessary.